tests/unit/modules/test_x509.py

```diff
@@ -165,12 +178,167 @@ def test_create_certificate(self):
-        self.assertIn(b'BEGIN RSA PRIVATE KEY', ret)
+    @skipIf(not HAS_M2CRYPTO, 'Skipping, M2Crypto is unavailble')
+    def test_create_crl(self):
+        ca_key = '''
+-----BEGIN RSA PRIVATE KEY-----
+MIICWwIBAAKBgQCjdjbgL4kQ8Lu73xeRRM1q3C3K3ptfCLpyfw38LRnymxaoJ6ls
+pNSx2dU1uJ89YKFlYLo1QcEk4rJ2fdIjarV0kuNCY3rC8jYUp9BpAU5Z6p9HKeT1
+2rTPH81JyjbQDR5PyfCyzYOQtpwpB4zIUUK/Go7tTm409xGKbbUFugJNgQIDAQAB
+AoGAF24we34U1ZrMLifSRv5nu3OIFNZHyx2DLDpOFOGaII5edwgIXwxZeIzS5Ppr
+yO568/8jcdLVDqZ4EkgCwRTgoXRq3a1GLHGFmBdDNvWjSTTMLoozuM0t2zjRmIsH
+hUd7tnai9Lf1Bp5HlBEhBU2gZWk+SXqLvxXe74/+BDAj7gECQQDRw1OPsrgTvs3R
+3MNwX6W8+iBYMTGjn6f/6rvEzUs/k6rwJluV7n8ISNUIAxoPy5g5vEYK6Ln/Ttc7
+u0K1KNlRAkEAx34qcxjuswavL3biNGE+8LpDJnJx1jaNWoH+ObuzYCCVMusdT2gy
+kKuq9ytTDgXd2qwZpIDNmscvReFy10glMQJAXebMz3U4Bk7SIHJtYy7OKQzn0dMj
+35WnRV81c2Jbnzhhu2PQeAvt/i1sgEuzLQL9QEtSJ6wLJ4mJvImV0TdaIQJAAYyk
+TcKK0A8kOy0kMp3yvDHmJZ1L7wr7bBGIZPBlQ0Ddh8i1sJExm1gJ+uN2QKyg/XrK
+tDFf52zWnCdVGgDwcQJALW/WcbSEK+JVV6KDJYpwCzWpKIKpBI0F6fdCr1G7Xcwj
+c9bcgp7D7xD+TxWWNj4CSXEccJgGr91StV+gFg4ARQ==
+-----END RSA PRIVATE KEY-----
+'''
+
+        ca_cert = x509.create_certificate(text=True,
+                                          signing_private_key=ca_key,
+                                          CN='Redacted Root CA',
+                                          O='Redacted',
+                                          C='BE',
+                                          ST='Antwerp',
+                                          L='Local Town',
+                                          Email='certadm@example.org',
+                                          basicConstraints="critical CA:true",
+                                          keyUsage="critical cRLSign, keyCertSign",
+                                          subjectKeyIdentifier='hash',
+                                          authorityKeyIdentifier='keyid,issuer:always',
+                                          days_valid=3650,
+                                          days_remaining=0)
+
+        with tempfile.NamedTemporaryFile('w+', delete=False) as ca_key_file:
+            ca_key_file.write(ca_key)
+            ca_key_file.flush()
+
+        with tempfile.NamedTemporaryFile('w+', delete=False) as ca_cert_file:
+            ca_cert_file.write(ca_cert)
+            ca_cert_file.flush()
+
+        with tempfile.NamedTemporaryFile('w+', delete=False) as ca_crl_file:
+            x509.create_crl(path=ca_crl_file.name,
+                            text=False,
+                            signing_private_key=ca_key_file.name,
+                            signing_private_key_passphrase=None,
+                            signing_cert=ca_cert_file.name,
+                            revoked=None,
+                            include_expired=False,
+                            days_valid=100,
+                            digest='sha512')
+
+        with salt.utils.fopen(ca_crl_file.name, 'r') as crl_file:
```

Comment:
This should use `salt.utils.files.fopen` instead. (The import above will need to be adjusted as well.)

Reply:
Hey @rallytime, fixed the call as you requested. I remember seeing it on this form on other files as well. lint runs with 10/10 and tests run on the local codebase. :)

I think the lint warnings should be updated though, because what made me use `salt.utils.fopen` was this lint warning here:

```
************* Module tests.unit.modules.test_x509
tests/unit/modules/test_x509.py:235: [W8470(resource-leakage), X509TestCase.test_create_crl] Resource leakage detected. Please use 'with salt.utils.fopen()' instead of 'with open()'. It assures salt does not leak file handles. 
```

Reply:
@jeduardo Ah, yes! Good catch. We didn't update those path names in the linter help messages.